extensions/aws/CMakeLists.txt

```diff
@@ -34,6 +34,9 @@ add_library(minifi-aws SHARED ${SOURCES})
 target_link_libraries(minifi-aws PUBLIC ${LIBMINIFI} Threads::Threads)
 
 target_wholearchive_library_private(minifi-aws AWS::aws-cpp-sdk-s3)
+if(NOT CMAKE_SYSTEM_PROCESSOR MATCHES "(x86)|(X86)")
```

Comment:
I actually made the change before I read Martin's comment, I thought it would be better to link the library on any non-x86 architecture as it may be needed there as well. But as I checked https://stackoverflow.com/questions/70475665/what-are-the-possible-values-of-cmake-system-processor there are more possibilities than I thought, like you said AMD64 also should have been included on windows. So it may be better to only concentrate on arm64 possibilities and we'll see in the future if we want to support more. I changed it to match only on arm64, ARM64, aarch64 and armv8 in 4037275310298711c7f949011b3e09e32a10ccdb